### PR TITLE
Config generator and better redis management

### PR DESCRIPTION
**Pull request by giuseppeM99**
- **Configuration**: Removed `config.py`, implemented _config generator_
- **Redis**: removed users hash
- **Redis**: now works with password

### CODE REVIEW

**Comment by giuseppeM99** [comment]
Ayyyy

**Comment by MarcoBuster** [comment]
@giuseppeM99 Thanks for your pull request!

File: start.py [hunk 1]
```diff
@@ -18,7 +18,57 @@
...
+    conf = conf + "\nBOT_TOKEN = \"" + token + "\"\n"
+    cfile = open("config.py", "w")
+    cfile.write(conf)
+    cfile.close()
+
 
 if __name__ == "__main__":
+    if not p.isfile("config.py"):
+        gen_config()
+    from src import main
```
Comment (MarcoBuster) [inline start.py]: Please move imports on top of the file.

File: config.sample.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 REDIS_HOST = 'localhost'
 REDIS_PORT = 6379
 REDIS_DB = 0
+REDIS_PASSWD = None
```
Comment (MarcoBuster) [inline config.sample.py]: Why `REDIS_PASSWD` instead of `REDIS_PASSWORD`?

File: start.py [hunk 3]
```diff
@@ -18,7 +18,57 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from src import main
+import os.path as p
+
+
+def gen_config():
+    conf = ('# Copyright (c) 2016-2017 Marco Aceti <dev@marcoaceti.it>\n'
```
Comment (MarcoBuster) [inline start.py]:
```python
foo = (
    "AAA"
    "BBB"
    "CCC"
)
```
It's better to see

File: start.py [hunk 4]
```diff
@@ -18,7 +18,57 @@
...
+    assert token, "You must insert a bot token"
+    conf = conf + "\nREDIS_HOST = \"" + rhost + "\""
+
+    conf = conf + "\nREDIS_PORT = " + str(rport)
+    conf = conf + "\nREDIS_DB = " + str(rdb)
+    if rpwd:
+        conf = conf + "\nREDIS_PASSWD = \"" + rpwd + "\""
+    else:
+        conf = conf + "\nREDIS_PASSWD = None"
+    conf = conf + "\n\n# Telegram Bot API token from t.me/BotFather"
```
Comment (MarcoBuster) [inline start.py]: Newline after and before if / else block please, it's better to see

File: requirements.txt [hunk 5]
```diff
@@ -1,2 +1,3 @@
 redis
 botogram
+dateutils
```
Comment (MarcoBuster) [inline requirements.txt]: Why this requirement?

**Review by MarcoBuster — CHANGES_REQUESTED** [review]
In summary it's okay, you need only fix some things.

**Review by MarcoBuster — APPROVED** [review]
#MarcoApproves

**Comment by MarcoBuster** [comment]
A last thing...

File: start.py [hunk 6]
```diff
@@ -55,12 +56,14 @@ def gen_config():
...
     conf = conf + "\nREDIS_PORT = " + str(rport)
     conf = conf + "\nREDIS_DB = " + str(rdb)
+
     if rpwd:
         conf = conf + "\nREDIS_PASSWD = \"" + rpwd + "\""
     else:
-        conf = conf + "\nREDIS_PASSWD = None"
-    conf = conf + "\n\n# Telegram Bot API token from t.me/BotFather"
-    conf = conf + "\nBOT_TOKEN = \"" + token + "\"\n"
+        conf += "\nREDIS_PASSWD = None"
```
Comment (giuseppeM99) [inline start.py]: You asked me to change from REDIS_PASSWD to REDIS_PASSWORD
I've changed it, you don't

**Comment by MarcoBuster** [comment]
Now we are _really_ ready to squash and **merge**!
Thanks again @giuseppeM99 for your **pull request** 👍

**Comment by MarcoBuster** [comment]
Elimina stronzo

**Comment by MarcoBuster** [comment]
@giuseppeM99 OH DEAR CELESTIA DELETE THIS FUCKING REVIEW

**Pull request merged**